w3af/core/controllers/chrome/js/dom_analyzer.js

```diff
@@ -865,6 +865,54 @@ var _DOMAnalyzer = _DOMAnalyzer || {
         return false;
     },
 
+    /**
+     * This is naive function which takes parentElement (the login form) and
+     * tries to find username input field within it.
+     * @param {Node} parentElement - parent element to scope to document.querySelectorAll()
+     * @param {String} exactSelector - optional CSS selector. If provided prevents
+     * using standard selectors
+     * @returns {NodeList} - result of querySelectorAll()
+     */
+    _getUsernameInput(parentElement, exactSelector = '') {
+        if (exactSelector) {
+            return document.querySelectorAll(exactSelector, parentElement);
+        }
+        result = document.querySelectorAll("input[type='email']", parentElement);
+        if (!result.length) {
+            result = document.querySelectorAll("input[type='text']", parentElement);
+        }
+        return result;
+    },
+
+    /**
+     * This is naive function which takes parentElement (the login form) and tries
+     * to find submit button within it.
+     * @param {Node} parentElement - parent element to scope to document.querySelectorAll()
+     * @param {String} exactSelector - optional CSS selector. If provided prevents
+     * using standard selectors
+     * @returns {NodeList} - result of querySelectorAll()
+     */
+    _getSubmitButton(parentElement, exactSelector = '') {
+        if (exactSelector) {
+            return document.querySelectorAll(exactSelector, parentElement);
+        }
+        result = document.querySelectorAll("input[type='submit']", parentElement);
+        if (!result.length) {
+            result = document.querySelectorAll("button[type='submit']", parentElement);
+        }
+        // Maybe it's just normal button with innerText: 'Login'...
+        if (!result.length) {
+            result = [];
+            let buttons = document.querySelectorAll('button', parentElement);
+            for (let button of buttons) {
+                if (button.innerText.toLocaleLowerCase().includes('log')) {
```

Comment:
I usually dislike code that is tied to a specific language. `log` might work for `login` and `logga` (which I believe is login in `.se`) but... for sites developed in other languages (spanish, polish, etc.) this will not work.

Also, `sign-in` will not be found.

Can we find an alternative?

Reply:
The only thing that comes into my mind is to return all buttons located inside `parentNode`. Hopefully there will be login button.